pkg/sql/sem/builtins/generator_probe_ranges.go

```diff
@@ -134,15 +134,26 @@ func makeProbeRangeGenerator(ctx *eval.Context, args tree.Datums) (eval.ValueGen
 			"only users with the admin role are allowed to use crdb_internal.probe_range",
 		)
 	}
+	// Trace the query to meta2. Return it as part of the error string if the query fails.
+	// This improves observability into a meta2 outage. We expect crdb_internal.probe_range
+	// to be available, unless meta2 is down.
+	meta2Ctx, sp := tracing.EnsureChildSpan(
```

Comment:
Preference to rewrite as follows, that way there is only ever one `ctx` in scope and it's the right one:

```go
var ranges []WhateverTypeThisIs
{
  ctx, sp := tracing.EnsureChildSpan(...)
  // ...
  var err error
  ranges, err = kvclient.ScanMetaKVs(meta2Ctx, ...)
  if err != nil { ... }
}
```